snoop/data/s3.py

```diff
@@ -421,10 +550,27 @@ def _data_mounted():
                 logger.warning('Failed to os.rmdir() the target directory %s (%s)', target, e)
 
         if not _pid_alive() and not _data_mounted() and not os.path.isdir(target):
-            tracer.count("umount_s3fs_success")
+            tracer.count("umount_success")
             return
 
         time.sleep(0.05 + 0.05 * retry)
 
-    tracer.count("umount_s3fs_failed")
-    raise RuntimeError(f'cannot remove old S3 mounts! target={target}, pid={pid}')
+    tracer.count("umount_failed")
+    raise RuntimeError(f'cannot remove old mounts! target={target}, pid={pid}')
+
+
+def mount_webdav(mount_args):
+    """Run subprocess to mount webdav share to target. Will wait until completed."""
+
+    cmd_bash = f"""
+    mount -t davfs http://10.66.60.1:9972{mount_args.get("webdav_url")} \\
+        -o conf={mount_args.get("configfile_path")} \\
+        {mount_args.get("target_path")} > {mount_args.get("logfile_path")} \\
+        2>&1 & echo $!
```

Comment:
Remove the `& echo $!` at the end, because this mount command will not be running in foreground, so we want to wait for it to finish.

You can set the pid to -1 (since we won't check for it's validity, see other comment in `adjust_mounts`